opentelemetry-sdk/src/metrics/meter_provider.rs

```diff
@@ -433,4 +434,28 @@ mod tests {
         // Now the shutdown should be invoked
         assert!(reader.is_shutdown());
     }
+
+    #[test]
+    fn same_meter_reused_same_scope() {
+        let provider = super::SdkMeterProvider::builder().build();
+        let _meter1 = provider.meter("test");
+        let _meter2 = provider.meter("test");
+        assert_eq!(provider.inner.meters.lock().unwrap().len(), 1);
+        let _meter3 =
+            provider.versioned_meter("test", Some("1.0.0"), Some("http://example.com"), None);
+        let _meter4 =
+            provider.versioned_meter("test", Some("1.0.0"), Some("http://example.com"), None);
+        let _meter5 =
+            provider.versioned_meter("test", Some("1.0.0"), Some("http://example.com"), None);
```

Comment:
Maybe also add one for schema URL? Not a blocker

Reply:
Yes, a ton of new tests will be added soon! 
There are more considerations for Meter identities.. not quite sure how to tackle them all. Adding basic tests now, just to be alerted myself while refactoring!